### ISSUE TITLE
Tolerances / Tension if 3D printed

### ISSUE THREAD

**Issue by etkaar** [adrianschlatter/threadlib#87]
Hi! What I do not understand so far: When I have a bolt and a nut, what about the tolerance? What is the correct approach? Just using _difference()_ will not cover tolerances, there might be to much tension.

I also tried _nut()_ and _bolt()_ (so int and ext), but there is too much tension if 3D printed.

**Comment by praetp** [comment]
Did you find a solution to this ?

**Comment by etkaar** [comment]
> Did you find a solution to this ?

I am now using this library:
https://github.com/BelfrySCAD/BOSL2

**Comment by adrianschlatter** [comment]
Indeed, at the moment, threadlib has clearances built in (in accordance with the standards, where applicable, but the clearances are not adjustable.

It may be possible to implement clearing classes A, B, … according to standards. But as OpenSCAD is not really a programming language (but more a file format), I‘m not motivated to try that.
